src/historywork/BatchDownloadVerifyBucketsWork.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "BatchDownloadVerifyBucketsWork.h"
+#include "VerifyBucketWork.h"
+#include "historywork/GetAndUnzipRemoteFileWork.h"
+#include "lib/util/format.h"
+
+namespace stellar
+{
+
+BatchDownloadVerifyBucketsWork::BatchDownloadVerifyBucketsWork(
```

Comment:
I am not sure this `BatchDownloadVerifyBucketsWork` implements the right contract:
I would expect it to be a batch of `<verify + download>` work (and therefore implement a generator function for verify+download)

Reply:
Also - why do we need a new class? I would expect `DownloadBucketsWork` to just be refactored (as it's already a class that "downloads and verifies buckets")